### DISCUSSION TITLE
Vapor/Octane: The /var/task/bootstrap/cache directory must be present and writable.

### DISCUSSION THREAD

**Discussion by dwightwatson** [livewire/livewire#3960] [7. Bugs (archived)] [answered]
I've been using Livewire with Vapor for a while. I opted-in to use [Octane on Vapor after it was announced earlier in the week](https://blog.laravel.com/vapor-octane-support-is-now-available). However I started receiving warnings on pages that rendered Livewire components:

```
Livewire\LivewireComponentsFinder::write
vendor/livewire/livewire/src/LivewireComponentsFinder.php:57
The /var/task/bootstrap/cache directory must be present and writable. (View: /var/task/resources/views/notes/create.blade.php)
```

Switching Octane off immediately resolved the problem again. I can see that @luischavez has done a heap of work improving Octane support (thank you!) but I'm not sure this specific use case is covered.

**Comment by joshhanley** [comment]
@dwightwatson not sure why Octane would cause those issues, but I just wanted to confirm, that you have modified Livewire's manifest_path config to support Vapor?

https://github.com/livewire/livewire/blob/ba5101a7b270d88991683c8ba0ce10d1ddf0c91d/config/livewire.php#L97-L111

**Comment by dwightwatson** [comment]
No - I hadn't, I wasn't aware of that configuration option. 

However, [digging into the PR that made this happen](https://github.com/livewire/livewire/pull/385/files), Livewire should automatically be inferring when it's running on Vapor and set this to the right value.

I wonder if perhaps the `SERVER_SOFTWARE` is different running under Octane?

```php
$isHostedOnVapor = $_ENV['SERVER_SOFTWARE'] ?? null === 'vapor';
```

---

Looks like there has been a number of issues in trying to keep up to date identifying with we're running on Vapor: 
* #1048
* #1051
* #1053

I wonder if it's worth seeing if the Laravel team could provide another environment variable that can be depended on to indicate that we're running on Vapor.

**Comment by georgeboot** [comment, marked as answer]
Update to the latest version of vapor-core. The issue has been fixed there.

**Reply by joshhanley** [reply]
@dwightwatson can you upgrade as George suggests and let us know if that fixes your issue?

**Reply by dwightwatson** [reply]
Upgraded from vapor-core 2.14.1 to 2.14.3 ([which includes this relevant fix](https://github.com/laravel/vapor-core/pull/101/files)).

However - now getting another issue that I'm not sure is directly related. The `/livewire/livewire.js` route returns a blank 200 response, which predictably results in a console error.

```
create:470 Uncaught ReferenceError: Livewire is not defined
```

Switching Octane off again immediately resolves the issue

**Reply by georgeboot** [reply]
For now, you could try publishing the assets via `artisan vendor:publish --tag=livewire:assets` so that the assets are served via the CDN, and not via the lambda function (which is anyway the preferred solution if you ask me).

**Reply by dwightwatson** [reply]
Thanks - I wasn't aware that was an option, but that makes a lot more sense.

Putting this as part of the Vapor deployment script fixes the issue and we're back in action.

I'd offer to make a PR for guidance deploying Livewire on Vapor, but `vendor:publish` is already documented so it's nothing new.

Thanks for the help!

**Reply by georgeboot** [reply]
Alternatively, `"@php artisan vendor:publish --force --tag=livewire:assets --ansi",` to your `post-autoload-dump` section of the `composer.json` instead of the `vapor.yml`, so that you use the published code both local and deployed.
